src/Compilers/VisualBasic/Test/Emit/Emit/EditAndContinue/EditAndContinueTestBase.vb

```diff
@@ -181,77 +181,117 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
-            AssertEx.Equal([handles], reader.GetEditAndContinueMapEntries().Where(Function(e) IsDefinition(e.Kind)), itemInspector:=AddressOf EncMapRowToString)
+            AssertEx.Equal(
+                [handles].Where(Function(h) h <> Nothing),
+                reader.GetEditAndContinueMapEntries().Where(Function(e) IsDefinition(e.Kind)), itemInspector:=AddressOf EncMapRowToString)
         End Sub
 
-        Friend Shared Sub CheckNames(reader As MetadataReader, [handles] As StringHandle(), ParamArray expectedNames As String())
+        Friend Shared Sub CheckNames(reader As MetadataReader, [handles] As IEnumerable(Of StringHandle), ParamArray expectedNames As String())
             CheckNames({reader}, [handles], expectedNames)
         End Sub
 
-        Friend Shared Sub CheckNames(readers As MetadataReader(), [handles] As StringHandle(), ParamArray expectedNames As String())
+        Friend Shared Sub CheckNames(readers As MetadataReader(), [handles] As IEnumerable(Of StringHandle), ParamArray expectedNames As String())
             Dim actualNames = readers.GetStrings([handles])
             AssertEx.Equal(expectedNames, actualNames)
         End Sub
 
-        Friend Shared Sub CheckNamesSorted(readers As MetadataReader(), [handles] As StringHandle(), ParamArray expectedNames As String())
+        Public Shared Sub CheckNames(readers As IList(Of MetadataReader), typeHandles As IEnumerable(Of TypeDefinitionHandle), ParamArray expectedNames As String())
+            CheckNames(readers, typeHandles, Function(reader, handle) reader.GetTypeDefinition(CType(handle, TypeDefinitionHandle)).Name, Function(handle) handle, expectedNames)
+        End Sub
+
+        Friend Shared Sub CheckNamesSorted(readers As MetadataReader(), [handles] As IEnumerable(Of StringHandle), ParamArray expectedNames As String())
             Dim actualNames = readers.GetStrings([handles])
             Array.Sort(actualNames)
             Array.Sort(expectedNames)
             AssertEx.Equal(expectedNames, actualNames)
         End Sub
 
+        Private Shared Sub CheckNames(Of THandle)(
+            readers As IList(Of MetadataReader),
+            entityHandles As IEnumerable(Of THandle),
+            getName As Func(Of MetadataReader, Handle, StringHandle),
+            toHandle As Func(Of THandle, Handle),
+            expectedNames As String())
+            Dim aggregator = GetAggregator(readers)
+
+            AssertEx.Equal(expectedNames, entityHandles.Select(
+                Function(handle)
+                    Dim typeGeneration As Integer
+                    Dim genEntityHandle = aggregator.GetGenerationHandle(toHandle(handle), typeGeneration)
+                    Dim nameHandle = getName(readers(typeGeneration), genEntityHandle)
+
+                    Dim nameGeneration As Integer
+                    Dim genNameHandle = CType(aggregator.GetGenerationHandle(nameHandle, nameGeneration), StringHandle)
```

Comment:
Can we use `DirectCast` here and below?

Reply:
Done.

Reply:
Doesn't work on handles as they are structs with custom conversion operators.